### ISSUE TITLE
[Feature] Transparent proxy support

### ISSUE THREAD

**Issue by Mygod** [Mygod/VPNHotspot#518]
Suggested through email: https://youtu.be/r6nXCgYkXTQ

**Comment by worstperson** [comment]
I was confused about what this request was, so I'm just going to leave notes here for the next time I read it :p

The video seems to show how to setup an Android device as a second gateway that routes incoming data through a VPN and configure the router to offer it to clients as the default gateway through it's DHCP server. So basically just setting a routing policy and masquerade rule rather than the normal tethering operation.

It's a clever solution for getting network wide VPN when you already have a good upstream (especially if that VPN is platform exclusive or too expensive to run on other hardware).

**Comment by Mygod** [comment]
Yes the idea is to share VPN with your neighboring nodes on the same network (by asking them to set your device as their gateway), instead of having a hotspot and have them connect to it. This is slightly more technical to set up and I consider this to be too niche and cumbersome to be useful. :)

**Comment by huzesama** [comment]
I'd still like you can update this feature if you can. As a "gateway sharing" feature
